build.gradle

```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
 jar {
     into "META-INF", { from generateClientPropertiesFile.clientPropsPath }
 }
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    sourceSets.all {
+        into(name + "/java", { from allJava })
+        into(name + "/resources", { from resources })
+    }
+}
+
+javadoc {
+    exclude("**/com/cloudant/tests/**",
+            "**/com/cloudant/test/**",
+            "**/resources/**")
+    include("**/lightcouch/DocumentConflictException.java",
+            "**/lightcouch/NoDocumentException.java",
+            "**/lightcouch/CouchDbException.java",
+            "**/client/**")
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourcesJar, javadocJar
+}
+
+//load signing and repository parameters from system properties
+['signing.keyId', 'signing.password', 'signing.secretKeyRingFile', 'ossrhUsername', 'ossrhPassword']
+        .each { propName ->
+    //set a property with the given name if the system property is set
+    //if the system property is not set then set the property to null if it isn't a signing one
+    if (System.properties.propName != null || !propName.startsWith("signing")) {
+        ext.(propName.toString()) = System.properties.propName
+    }
+}
+
+signing {
+    //only apply signing when it is a release and is being published
+    required {
+        isReleaseVersion && gradle.taskGraph.hasTask("uploadArchives")
+    }
+    //when signing, sign the archives
+    sign configurations.archives
+}
+
+uploadArchives {
+    repositories {
+        mavenDeployer {
+
+            //when publishing sign the pom
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            //augment the pom with additional information
+            pom.project {
+                packaging 'jar'
+                name project.clientName
+                description 'official Cloudant Client for Java'
```

Comment:
I copy/pasted from the previous pom without checking, I guess "Client" should probably be "client" too